stdlib/public/core/UnsafePointer.swift.gyb

```diff
@@ -429,6 +429,11 @@ public struct ${Self}<Pointee>
     return UnsafeMutablePointer(rawPtr)
   }
 
+  @available(*, unavailable, message: "Swift currently only supports freeing entire heap blocks, use deallocate() instead")
```

Comment:
as explained in the [proposal](https://github.com/kelvin13/swift-evolution/blob/9d305ced7b82b9cf5854b55b3f0d08952853d046/proposals/0184-improved-pointers.md#source-compatibility) we are preferring a hard `unavailable` over a soft deprecation.